cmd/frontend/graphqlbackend/graphqlutil/connection_resolver_test.go

```diff
@@ -0,0 +1,555 @@
+package graphqlutil
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/graph-gophers/graphql-go"
+
+	"github.com/sourcegraph/sourcegraph/internal/database"
+)
+
+var TOTAL_COUNT = int32(10)
+
+type TestConnectionNode struct {
+	id int
+}
+
+func (n TestConnectionNode) ID() graphql.ID {
+	return graphql.ID(fmt.Sprintf("%d", n.id))
+}
+
+type TestConnectionStore struct {
+	t                  *testing.T
+	TestComputeNodes   func(ctx context.Context, args *database.PaginationArgs)
+	ComputeTotalCalled int
+	ComputeNodesCalled int
+}
+
+func (s *TestConnectionStore) ComputeTotal(ctx context.Context) (*int32, error) {
+	s.ComputeTotalCalled = s.ComputeTotalCalled + 1
+	total := TOTAL_COUNT
+
+	return &total, nil
+}
+
+func (s *TestConnectionStore) ComputeNodes(ctx context.Context, args *database.PaginationArgs) ([]*TestConnectionNode, error) {
+	s.ComputeNodesCalled = s.ComputeNodesCalled + 1
+	nodes := []*TestConnectionNode{{id: 0}, {id: 1}}
+
+	s.TestComputeNodes(ctx, args)
+
+	return nodes, nil
+}
+
+func (*TestConnectionStore) MarshalCursor(n *TestConnectionNode) (*string, error) {
+	cursor := string(n.ID())
+
+	return &cursor, nil
+}
+
+func (*TestConnectionStore) UnMarshalCursor(cursor string) (*int32, error) {
+	num, err := strconv.Atoi(cursor)
+	if err != nil {
+		return nil, err
+	}
+
+	id := int32(num)
+
+	return &id, nil
+}
+
+func TestConnectionTotalCount(t *testing.T) {
+	ctx := context.Background()
+	store := &TestConnectionStore{t, func(context.Context, *database.PaginationArgs) { return }, 0, 0}
```

Comment:
```suggestion
	store := &TestConnectionStore{t, func(context.Context, *database.PaginationArgs) { }, 0, 0}
```

Also: you should name the fields. That helps readability and it allows you to use default values here.

```suggestion
	store := &TestConnectionStore{t: t, ComputeNodes: func(context.Context, *database.PaginationArgs) { }}
```